src/api/result_proxy_api_client.rs

```diff
@@ -0,0 +1,324 @@
+use reqwest::blocking::Client;
+use serde::{Deserialize, Serialize};
+
+const EMPTY_HEX_STRING_32: &str =
+    "0x0000000000000000000000000000000000000000000000000000000000000000";
+const EMPTY_WEB3_SIG: &str = "0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000";
+
+/// Represents a computation result that can be uploaded to IPFS via the iExec result proxy.
+///
+/// This struct encapsulates all the necessary information about a completed computation task
+/// that needs to be stored permanently on IPFS. It includes task identification, metadata,
+/// the actual result data, and cryptographic proofs of computation integrity.
+///
+/// The struct is designed to be serialized to JSON for transmission to the result proxy API,
+/// with field names automatically converted to camelCase to match the expected API format.
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ResultModel {
+    /// Unique identifier of the task on the blockchain
+    pub chain_task_id: String,
+    /// Unique identifier of the deal this task belongs to
+    pub deal_id: String,
+    /// Index of the task within the deal
+    pub task_index: u32,
+    /// Compressed result data as a byte array
+    pub zip: Vec<u8>,
+    /// Cryptographic hash of the computation result
+    pub deterministic_hash: String,
```

Comment:
Typo here, it must be aligned with Java implementation
```suggestion
    pub determinist_hash: String,
```

Reply:
Resolved in [613bfb8](https://github.com/iExecBlockchainComputing/tee-worker-post-compute-rust/pull/16/commits/613bfb87987404f3ff266d831255227b3c906097)